packages/next-bundle-analyzer/index.js

```diff
@@ -1,19 +1,26 @@
 module.exports =
-  ({ enabled = true, openAnalyzer = true } = {}) =>
+  (bundleAnalyzerConfig = {}) =>
   (nextConfig = {}) => {
     return Object.assign({}, nextConfig, {
       webpack(config, options) {
-        if (enabled) {
+        const analyzerConfig =
+          typeof bundleAnalyzerConfig === 'function'
+            ? bundleAnalyzerConfig(config, options)
+            : Object.assign(
```

Comment:
I used `Object.assign` here instead of spread since it was already in use above - that said, Spread should be able to be used since it has been available since Node 8 and NextJS has [an engines setting for Node set to `>=12.22.0`](https://github.com/vercel/next.js/blob/canary/package.json#L215)